aws/data_source_aws_ec2_instance_type.go

```diff
@@ -0,0 +1,440 @@
+			},
+			"maximum_ipv4_addresses_per_interface": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"maximum_ipv6_addresses_per_interface": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"ipv6_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"ena_support": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"gpus": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_gpu_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"fpgas": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_fpga_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"supported_placement_strategies": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"accelerators": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"hibernation_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"burstable_performance_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"dedicated_hosts_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"auto_recovery_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEc2InstanceTypeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	params := &ec2.DescribeInstanceTypesInput{}
+
+	if v, ok := d.GetOk("instance_type"); ok {
+		params.InstanceTypes = []*string{aws.String(v.(string))}
+	}
```

Comment:
For a required argument, this is probably unnecessary. Also, it will be handy for error messages to have the instance type in a variable.

```suggestion
	instanceType := d.Get("instance_type").(string)
	params.InstanceTypes = []*string{aws.String(instanceType)}
```